packages/block-library/src/table/state.js

```diff
@@ -26,105 +28,186 @@ export function createTable( {
+	} = selection;
+
+	return mapValues( tableSections, ( section, sectionName ) => {
+		if ( selectionSectionName && selectionSectionName !== sectionName ) {
+			return section;
+		}
+
+		return section.map( ( row, rowIndex ) => {
+			if ( selectionRowIndex && selectionRowIndex !== rowIndex ) {
 				return row;
 			}
 
 			return {
-				cells: row.cells.map( ( cell, currentColumnIndex ) => {
-					if ( currentColumnIndex !== columnIndex ) {
-						return cell;
+				cells: row.cells.map( ( cellAttributes, columnIndex ) => {
+					const cellLocation = {
+						sectionName,
+						columnIndex,
+						rowIndex,
+					};
+
+					if ( ! isCellSelected( cellLocation, selection ) ) {
+						return cellAttributes;
 					}
 
-					return {
-						...cell,
-						content,
-					};
+					return updateCell( cellAttributes );
 				} ),
 			};
-		} ),
-	};
+		} );
+	} );
+}
+
+/**
+ * Returns whether the cell at `cellLocation` is included in the selection `selection`.
+ *
+ * @param {Object} cellLocation An object containing cell location properties.
+ * @param {Object} selection    An object containing selection properties.
+ *
+ * @return {boolean} True if the cell is selected, false otherwise.
+ */
+export function isCellSelected( cellLocation, selection ) {
+	if ( ! cellLocation || ! selection ) {
+		return false;
+	}
+
+	switch ( selection.type ) {
+		case 'column':
+			return selection.type === 'column' && cellLocation.columnIndex === selection.columnIndex;
+		case 'cell':
+			return selection.type === 'cell' &&
+				cellLocation.sectionName === selection.sectionName &&
+				cellLocation.columnIndex === selection.columnIndex &&
+				cellLocation.rowIndex === selection.rowIndex;
+	}
 }
 
 /**
  * Inserts a row in the table state.
  *
- * @param {Object} state            Current table state.
- * @param {string} options.section  Section in which to insert the row.
- * @param {number} options.rowIndex Row index at which to insert the row.
+ * @param {Object} state                Current table state.
+ * @param {string} options.sectionName  Section in which to insert the row.
+ * @param {number} options.rowIndex     Row index at which to insert the row.
  *
  * @return {Object} New table state.
  */
 export function insertRow( state, {
-	section,
+	sectionName,
 	rowIndex,
 	columnCount,
 } ) {
-	const cellCount = columnCount || state[ section ][ 0 ].cells.length;
+	const firstRow = getFirstRow( state );
+	const cellCount = columnCount || get( firstRow, [ 'cells', 'length' ], 2 );
 
 	return {
-		[ section ]: [
-			...state[ section ].slice( 0, rowIndex ),
+		[ sectionName ]: [
+			...state[ sectionName ].slice( 0, rowIndex ),
 			{
-				cells: times( cellCount, () => ( {
-					content: '',
-					tag: section === 'head' ? 'th' : 'td',
-				} ) ),
+				cells: times( cellCount, ( index ) => {
+					const firstCellInColumn = get( firstRow, [ 'cells', index ], {} );
+					const inheritedAttributes = pick( firstCellInColumn, INHERITED_COLUMN_ATTRIBUTES );
```

Comment:
Why is alignment inherited?

Reply:
When adding a new row, the newly added cells will keep the same alignment as the rest of their adjacent column.

As already mentioned in this PR, it seems like all cells in a column should have the same alignment, so this maintains that experience for the user. In my testing it felt like the right way to go.

Reply:
Sounds good